packages/clerk-js/src/ui/components/OrganizationProfile/DomainList.tsx

```diff
@@ -21,6 +21,55 @@ type DomainListProps = GetDomainsParams & {
   fallback?: React.ReactNode;
 };
 
+const useDomainList = () => {
+  const { isAuthorizedUser: canDeleteDomain } = useGate({ permission: 'org:sys_domains:delete' });
+  const { isAuthorizedUser: canVerifyDomain } = useGate({ permission: 'org:sys_domains:manage' });
+
+  return {
+    showDotMenu: canDeleteDomain || canVerifyDomain,
+    canVerifyDomain,
+    canDeleteDomain,
+  };
+};
+
+const DomainListDotMenu = ({
+  redirectSubPath,
+  domainId,
+}: Pick<DomainListProps, 'redirectSubPath'> & {
+  domainId: OrganizationDomainResource['id'];
+}) => {
+  const { navigate } = useRouter();
+  const { canDeleteDomain, canVerifyDomain } = useDomainList();
+
+  return (
+    <ThreeDotsMenu
+      actions={[
+        ...(canVerifyDomain
```

Comment:
🙃 Consider building the actions array outside JSX for improved readability using a `buildActions` function that handles the business logic.

Reply:
I created a hook that returns the actions, the conditional logic depended too much on the state + router.